feat: DAQ Heartbeat

nicholasching opened the pull request:
Completed basic heartbeat functionality for DAQ with the following features and design considerations:

Decided to implement heartbeat on separate heartbeat.go file for modularization.

On an interval, currently 3s, send POST request to the server with the following info:
- timestamp
- vehicle_ID - currently retrieved from env variable; i was thinking this could be used as a tag to filter data in the future. For example, we can configure the vehicle with different identifiers to differentiate between development, testing, practice, competition, etc, and then filter runs by vehicle_ID
- session_ID - unique hash generated on each startup of the heartbeat script, allows us to track different test runs and associate CAN frame data w/each run, maybe some sort of list view for runs in the future.
- CAN connection state for basic debug if the CAN telemetry does not work, but heartbeat does.

If communication fails, nothing happens; we will continue sending on the same regular interval.

TylerStAmour committed:
daq: basic boilerplate setup

untested

TylerStAmour committed:
daq: send/enqueue template

ManushPatell committed:
Suspension Potentiometer calculation

* added positionfromV function and implemented in LV Controller

* fixed unused variable and declaration

* format issues

* typo

* fix formatting

* maybe now its for-real

* apply clang-format fixes

* fixing suspension formatting

* restore platformio file

* changes on dbc file for suspension travel

* updated namespace

* updated test cases and edge case

* naming and added comment

* fix build issue

BlakeFreer committed:
Add windows symlink fix (#528)

BlakeFreer committed:
cantools scripts (#530)

TylerStAmour committed:
basic caching mechanism for can frames

Writes them to a local SQLite DB for now

luaibash committed:
Convert TMS to use FreeRTOS (#527)

* Added configurations for tms to use FreeRTOS

* Converted main.cc to use FreeRTOS tasks

* Removed lib_deps from env in platformio file for tms

* generated files from cubemx

TylerStAmour committed:
daq: interval between repeated frames + handle both CAN busses

Plus some other misc work

BlakeFreer committed:
increase speed limit to 75 kmh (#537)

SpeakBoy committed:
Implement static code analysis on projects (#534)

* Add static code analysis to dashboard project

* Add static code analysis to front_controller

* Add static code analysis to lvcontroller

* Add static code analysis to tms

* Revert check_severity to default value

* Add static code analysis to demo projects

* Add pio check to build-test-projects

* Add pio check to pre-commit

ManushPatell committed:
fc suspension  (#536)

* added fc suspension on dbc

* change class to sustravel12

* called in main.cc and fixed include statement

* inserted in loop

SpeakBoy committed:
Improve static code analysis time for PRs (#542)

* Check for changes only in certain directories

* Fix static code analysis workflow

* Fix build-test-projects.yml

* Add test files for PR analysis

* Remove test files from repo

* Reduce number of pre-commit checks

* Reduce number of PR analysis checks

luaibash committed:
Transmit Git Hash over CAN (#540)

* Added generate git hash script

* Added platformio configs to use generate_git_hash.py as a new pre script file

* Added sending of git hash in lvcontroller

* Added 1hz task and git hash send to front controller

* Added 1hz task and git hash send to tms

* Used env variable for finding include directory of project

nicholasching committed:
feat: completed heartbeat constructor

nicholasching committed:
feat: completed heartbeat constructor

nicholasching committed:
feat: completed heartbeat implementation

nicholasching committed:
feat: completed heartbeat implementation

nicholasching committed:
fix: heartbeat types

nicholasching committed:
fix: heartbeat types

nicholasching committed:
feat: integrated heartbeat into daq main

nicholasching committed:
feat: integrated heartbeat into daq main

sokosam commented on scripts/daq/main.go:
```diff
@@ -41,13 +43,22 @@ func main() {
+				logger.Error("Failed to send heartbeat", zap.Error(err))
+			}
 		}
 	}
```
Can we name both to Interval? Feels weird to have one be uploadTimer and heartbeatInterval
```suggestion
	uploadInterval := time.NewTimer(time.Second)

	heartbeatInterval := time.NewTicker(3 * time.Second)
	defer heartbeatInterval.Stop()

	for {
		select {
		case <-uploadInterval.C:
			err = telemetry.Upload()
			if err != nil {
				fmt.Printf("failed to upload telemetry data: %v\n", err)
			}
		case <-heartbeatInterval.C:
			err = heartbeat.SendHeartbeat()
			if err != nil {
				logger.Error("Failed to send heartbeat", zap.Error(err))
			}
		}
	}

```

sokosam reviewed (commented):
Rest LGTM

BlakeFreer committed:
cobs library (#547)

BlakeFreer committed:
rename cobs library (#548)

BlakeFreer committed:
Disable pre-commit static analysis (#549)

Takes way to long. We should not be checking static analysis in
precommit since it takes several minutes.

This should only be a github check

samparent97 committed:
Remove samparent97 from CODEOWNERS (#554)

BlakeFreer committed:
Remove the CODEOWNERs file (#556)

PRs can now be approved by anyone in the organization.

noahjaye committed:
Sockets for C++ SIL (#551)

* Clear out mcal sil. start sockets

* Adding Cobs message framing.`

* Send data with COBS

* Fix flags in platformio

* Fix Nanopb bug with PlatformIO

See the comment in `platformio.ini`

The bug is described in https://github.com/nanopb/nanopb/issues/818

* Add pwm to protobuf

---------

Co-authored-by: Blake Freer <bkfreer10@gmail.com>

nicholasching committed:
feat: rpi logger

nicholasching committed:
feat: rpi logger

HarshPatel08 committed:
Configurable pedal-to-torque mapping (#560)

* Add configurable pedal-to-torque mapping

* Fix clang-format issues in tuning.hpp

* Set default pedal map to linear

noahjaye committed:
Fix the dashboard (#561)

* Migrate dashboard to LVGL 9.3 and add StatusBar

- Update LVGL dependency from 8.3 to 9.3 for both Linux and STM32
- Migrate display drivers to LVGL 9.3 API (lv_display_t, flush callbacks)
- Regenerate FE_Logo with RGB565 format for LVGL 9.3 compatibility
- Fix STM32 build: disable ARM Helium/NEON SIMD in lv_conf.h
- Fix exception handling for embedded targets (STM32)
- Add development tooling and documentation

Tested: Both targets build successfully

* Address review feedback: memory safety, config updates, and code cleanup

* Removed logo splash page. Saved 100KB in Flash

* Remove unused submodules :)

* Fix LVGL config issues

Dashboard wasn't showing anything:
- enabled LTDC callback in cubemx
- told platformio not to override the hal_conf with a build config
  option
- enabled st ltdc in lvgl.h
- enabled dma in ltdc

Still to fix:

fonts don't show up correctly

* Fix the dashboard, cubemx overwrites stm32f4xx_it.c which is an issue and must be kept in mind

* fix overwriting stm32f44x_it.c in cubemx and reset pin config

* format

* Finalize push

* format & pin out config

* repeat

---------

Co-authored-by: Furqaan <furqaankhurram@gmail.com>
Co-authored-by: Blake Freer <bkfreer10@gmail.com>
Co-authored-by: ManushPatell <manush.patelll@gmail.com>

ManushPatell committed:
test logger & influx db

ManushPatell committed:
test logger & influx db

ManushPatell committed:
influx-db

ManushPatell committed:
influx-db

ManushPatell committed:
Merge branch 'daq-nick-heartbeat' of https://github.com/macformula/racecar into daq-nick-heartbeat

ManushPatell committed:
grafana dashboard json file

ManushPatell committed:
include pt.dbc for powertrain can parsing

ManushPatell committed:
testing logger in linux-vm

ManushPatell committed:
logger.py issues, need to fix on vm

ManushPatell committed:
.

ManushPatell committed:
README.md, sh files, .deb file is too large :(

ManushPatell committed:
clang format ignore file

ManushPatell committed:
remove clang format file

ManushPatell committed:
clang format is annoying me

ManushPatell committed:
revert to old config

ManushPatell committed:
update erm

ManushPatell committed:
pls pls pls fix

ManushPatell committed:
wow i hate this

ManushPatell committed:
update README.md

noahjaye committed:
exclude stm32 from format checks

noahjaye committed:
Examine bindings failure

ManushPatell committed:
.sh file for influxdb cloud, minimal setup

ManushPatell committed:
Merge branch 'daq-nick-heartbeat' of https://github.com/macformula/racecar into daq-nick-heartbeat

ManushPatell committed:
add APPS and BPPS

ManushPatell committed:
Update LV FW (#567)

* clean branch, update lv pinout config

* new bindings

* fix hsd binding

* read channels periph

* hsd classes

* hsd class implementation

* add cli testing, and fix build issue

* add bindings for stm32

* fix clang format

* dashboard clang format

* sil bindings, will need to improve after comp

* update dcdc, and cli bindings

* sil bindings, still throw errors

* dcdc

* add new dcdc can message in veh bus

* add missing channel

ManushPatell committed:
add thresholds and new fields

ManushPatell committed:
lv controller updates, add mapping (#573)

noahjaye committed:
FC Rev4 changes (#569)

* Update fc pinout

* pinouts fc new rev

* new hsd folder

* add to alerts, new signal name for overcurrent

* fixed include

* fix hsd error

* add veh can to hsd update

* fix return

* hsd

* fix bindings

* fix another build error

* adding mcal bindings

* adding mcal bindings

* add thresholds, and mappign

* .

---------

Co-authored-by: ManushPatell <manush.patelll@gmail.com>

ManushPatell committed:
bypass logo screen (#574)

* bypass logo screen

* fix logic

ManushPatell committed:
Merge branch 'main' into daq-nick-heartbeat

ManushPatell committed:
dont need dashboard changes, restore them

ManushPatell committed:
Merge branch 'daq-nick-heartbeat' of https://github.com/macformula/racecar into daq-nick-heartbeat

ManushPatell committed:
another unneccesary change to dash

ManushPatell committed:
tuning apps & bpps (#577)

ManushPatell committed:
make influx optional

ManushPatell committed:
Merge branch 'main' into daq-nick-heartbeat

ManushPatell committed:
remove message

ManushPatell committed:
Merge branch 'daq-nick-heartbeat' of https://github.com/macformula/racecar into daq-nick-heartbeat
